### PR TITLE
stash the build logs before the test fails

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@stevekuznetsov ptal, this will ensure we actually get the build log output for failures like: https://github.com/openshift/origin/issues/13046#event-972482696

right now when the build fails, we bomb out and never get the logs (in theory the build pod's container logs should exist, but i haven't seen them in the s3 artifacts for whatever reason)

**Comment by bparees** [comment]
[test]

File: test/end-to-end/core.sh [hunk 1]
```diff
@@ -420,8 +420,8 @@ os::cmd::try_until_text "oc get builds --namespace test -o jsonpath='{.items[0].
 BUILD_ID="$( oc get builds --namespace test -o jsonpath='{.items[0].metadata.name}' )"
 # Ensure that the build pod doesn't allow exec
 os::cmd::expect_failure_and_text "oc rsh ${BUILD_ID}-build" 'forbidden'
-os::cmd::try_until_text "oc get builds --namespace test -o jsonpath='{.items[0].status.phase}'" "Complete" "$(( 10*TIME_MIN ))"
-os::cmd::expect_success "oc logs build/${BUILD_ID} --namespace test  > '${LOG_DIR}/test-build.log'"
+os::cmd::expect_success "oc logs build/${BUILD_ID} --namespace test  -f > '${LOG_DIR}/test-build.log'"
+os::cmd::try_until_text "oc get builds --namespace test -o jsonpath='{.items[0].status.phase}'" "Complete" "$(( 1*TIME_MIN ))"
```
Comment (stevekuznetsov) [inline test/end-to-end/core.sh]: doesn't need to be a try_until anymore

Comment (bparees) [inline test/end-to-end/core.sh]: i'm not totally confident the build status will be updated as soon as the log finishes.

Comment (stevekuznetsov) [inline test/end-to-end/core.sh]: Seems weird but ok. This LGTM

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/10773
[test]

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
[merge]

**Pull request merged**